packages/@aws-cdk/aws-pipes-alpha/lib/pipe.ts

```diff
@@ -233,6 +233,10 @@ export class Pipe extends PipeBase {
      */
     const source = props.source.bind(this);
     props.source.grantRead(this.pipeRole);
+    if (SourceWithDeadLetterTarget.isSourceWithDeadLetterTarget(props.source)) {
+      props.source.grantPush(this.pipeRole, props.source.deadLetterTarget);
+    }
```

Comment:
Can you add some comments to explain why this grantPush permission is needed?

Reply:
I added some notes in https://github.com/aws/aws-cdk/pull/29476/commits/ecba4fa2c25d57cfd8d0877f7573c198e0993d47.